tests/main/xdg-open-portal/task.yaml

```diff
@@ -52,19 +52,18 @@ prepare: |
     EOF
 
     # Set up file type handlers
-    mkdir -p ~test/.config
-    cat << EOF > ~test/.config/mimeapps.list
+    session-tool -u test mkdir -p ~test/.config
+    session-tool -u test sh -c 'cat > ~test/.config/mimeapps.list' << EOF
     [Default Applications]
     x-scheme-handler/http=test-browser.desktop
     text/plain=test-editor.desktop
     EOF
 
     # Prepare test data for editor
-    commondir=~test/snap/test-snapd-desktop/common
-    testfile="$commondir/test.txt"
-    mkdir -p "$commondir"
-    echo "Hello World" > "$testfile"
-    chown -R test:test ~test/snap
+    session-tool -u test mkdir -p ~test/snap/test-snapd-desktop/common
+    session-tool -u test sh -c 'cat > ~test/snap/test-snapd-desktop/common/test.txt' << EOF
+    Hello World
+    EOF
```

Comment:
no preference, but you could have done the same thing without using a heredoc but instead using tee, running under session-tool, no ?

Reply:
Yes but I can never remember how to use tee correctly. I just followed the pattern used for the other files in the set of related tests.

Reply:
Hmm `session-tool -u test sh -c 'echo "foo" > ~test/snap/test-snapd-desktop/common/test.txt'` would work too